### DISCUSSION TITLE
taxid usage

### DISCUSSION THREAD

**Discussion by phiweger** [kblin/ncbi-genome-download#182] [Q&A]
How does one specidy the `--taxid` argument correctly? When I call ...

```
ncbi-genome-download --taxid 83654 bacteria
```

... I get ...

```
ERROR: No downloads matched your filter. Please check your options.
```

However, this works:

```
ncbi-genome-download --genus leclercia bacteria
```

However, I would like to download the entire family, so using the taxid would be much more convenient.

Thank you,
Adrian

**Comment by kblin** [comment]
The `--taxid`/`--species-taxid` parameters are limited to the taxa explicitly mentioned in the assembly summary file from NCBI. `--taxid` is the ID associated to the exact organism the sequence belongs to, and the other one is the ID of the parent species. No other levels are available for `ncbi-genome-download` while running because the tool only checks the NCBI FTP server and does not talk to the taxonomy database.

Having said that, @jrjhealey recently contributed an additional script that uses the ETE toolkit to scan through the NCBI taxonomy to find the taxids of all descendant taxa. See https://github.com/kblin/ncbi-genome-download/blob/master/contrib/gimme_taxa.py

That script will create a file with all the taxids that you want to download using `ncbi-genome-download`.

Hope this helps.
